src/Microsoft.AspNet.Mvc.Core/DefaultControllerActionArgumentBinder.cs

```diff
@@ -43,18 +44,17 @@ public class DefaultControllerActionArgumentBinder : IControllerActionArgumentBi
                         nameof(actionContext));
             }
 
+
+            var methodParameters = actionDescriptor.MethodInfo.GetParameters();
             var parameterMetadata = new List<ModelMetadata>();
             foreach (var parameter in actionDescriptor.Parameters)
             {
+                var parameterInfo = methodParameters.Where(p => p.Name == parameter.Name).Single();
                 var metadata = _modelMetadataProvider.GetMetadataForParameter(
-                    methodInfo: actionDescriptor.MethodInfo,
```

Comment:
It's simpler for caching this way. The problem is that getting a `Type` is easy and cheap. So something like `GetMetadataForType` or `GetMetadataForProperties(Type)` can be called by anyone at minimal expense.

However you care about `MethodInfo` or `ParameterInfo` then you **need** to care about caching. So you might as well go all the way and get the `ParameterInfo`. I did consider a model where you return all the parameters for a method, but it felt kinda zero-sum to me.

I still want to eliminate `GetMetadataForParameter` asap.